src/test/optimizer/greedy_operator_ordering_test.cpp

```diff
@@ -147,10 +149,9 @@ TEST_F(GreedyOperatorOrderingTest, HyperEdges) {
         PredicateNode::make(equals_(add_(b_a, c_a), d_a),
           JoinNode::make(JoinMode::Inner, equals_(b_a, c_a),
             node_b,
-            PredicateNode::make(less_than_equals_(c_a, d_a),
-              JoinNode::make(JoinMode::Inner, equals_(c_a, d_a),
+              JoinNode::make(JoinMode::Inner, join_predicates,
```

Comment:
again, with `expression_vector` you could put the two predicates here directly without having to create a temporary variable